### ISSUE TITLE
dotnet publish with custom output directory breaks Blazor wasm App

### ISSUE THREAD

**Issue by DominikPf** [dotnet/sdk#12372]
If i run dotnet publish without a specific output Directory everything works fine.
> dotnet publish .\MyApp.sln --configuration Release

---------

If i use a custom Output Directory e.g. _artifacts_, the App crashes.

> dotnet publish .\MyApp.sln --configuration Release --output artifacts

The breaking Part is the rewrite section.
If i remove the rewrite section everything works fine.
The mimeMaps and DynamicTypes declaration are also new, but they do not break the App.

-------- 
It seems that the cli creates different web.config for these two cases (see Attachments).

[web configs.zip](https://github.com/dotnet/sdk/files/4883373/web.configs.zip)

SDK Version: 3.1.301

**Comment by timheuer** [comment]
Is the solution only a Blazor Wasm project in it?  Using the default template both commands emitted the same web.config for me.  /cc @captainsafia @mkArtakMSFT -- this should probably move to your repo

**Comment by DominikPf** [comment]
You can reproduce the Issue with a new Blazor wasm (aspnetcore hosted) template.

> dotnet new blazorwasm -ho

> dotnet publish .\MyApp.sln -o artifacts

I think i found the origin of thi behavior:
It seems that the two web.configs (from the server and the client project) are merged. (Not sure if this is a real issue or expected for multiple projects)

If i publish the server csproj everything works fine.

**Comment by timheuer** [comment]
@DominikPf Yes, when you choose the hosted option, the assets to publish are from the Server project, which at build/publish time will also include the Client application.  In the hosted scenario, the rewrites and such are handled on the ASP.NET server app and thus the web.config settings you see for just the client app are not needed.  This is the design.  The docs for hosting are here that attempt to explain that: https://docs.microsoft.com/en-us/aspnet/core/blazor/host-and-deploy/webassembly?view=aspnetcore-3.1#hosted-deployment-with-aspnet-core

/cc @danroth27
